cub/cub/block/block_scan.cuh

```diff
@@ -2242,6 +2242,67 @@ public:
     }
   }
 
+  //! @rst
+  //! Computes an inclusive block-wide prefix scan using the specified binary ``scan_op`` functor.
+  //! Each thread contributes an array of consecutive input elements.
+  //!
+  //! - Supports non-commutative scan operators.
+  //! - @blocked
+  //! - @granularity
+  //! - @smemreuse
+  //!
+  //! Snippet
+  //! +++++++
+  //!
+  //! The code snippet below illustrates an inclusive prefix max scan of 512 integer items that
+  //! are partitioned in a :ref:`blocked arrangement <flexible-data-arrangement>` across 128 threads
+  //! where each thread owns 4 consecutive items.
+  //!
+  //! .. literalinclude:: ../../test/catch2_test_block_scan_api.cu
+  //!
+  //!     :language: c++
+  //!     :dedent:
+  //!     :start-after: example-begin inclusive-scan-array-init-value
+  //!     :end-before: example-end inclusive-scan-array-init-value
+  //!
+  //! Suppose the set of input ``thread_data`` across the block of threads is
+  //! ``{[0, -1], [2, -3],[4, -5], ... [126, -127]}``.
+  //! The corresponding output ``thread_data`` in those threads will be
+  //! ``{[1, 1], [2, 2],[3, 3], ... [126, 126]}``.
```

Comment:
suggestion: I think we cover this in the kernel itself. I think seeing input and expected output in the docs might be confusing, but feel free to disagree:
```suggestion
```